pipeline.py

```diff
@@ -0,0 +1,72 @@
+import pmu,common
+from tabulate import tabulate
+import itertools,sys,re
+
+def pipeline_view(log,depths):
+  out_pipe=open(log.replace('csv','pipeline.log'),'w');
+  pipe_list=get_log(log)
+  event_list=[]
+  depth_list=[]
+  lines=[]
+  for i in depths:
+    event_list.append(depths[i][0])
+    depth_list.append(int(depths[i][1]))
+  issued_name=depths['issued'][0] 
+  issued_list=[i for i,item in enumerate(event_list) if re.search(issued_name,item)]
+  issued_index=issued_list[0]
+  lines=get_search_list(pipe_list,event_list[0])
+  rows=int(len(lines)/depth_list[0]) 
+  indices=[i+1 for i in range(max(depth_list))]
+  chunks=[[[0 for i in range(l)] for j in range(rows)] for k,l in zip(range(len(event_list)),depth_list)]
+  processed_chunks=[[[0 for i in range(l)] for j in range(rows)] for k,l in zip(range(len(event_list)),depth_list)]
+  percent_chunks=[[[0 for i in range(l)] for j in range(rows)] for k,l in zip(range(len(event_list)),depth_list)]
+  final_percent=[[[0 for i in range(k)] for j,k in zip(range(len(event_list)),depth_list)] for l in range(rows)]
+  totals=[[0 for i in range(len(event_list))] for j in range(rows)]
+  for event_name,event_depth in zip(event_list,depth_list):
+    sum=0
+    lines=[]
+    index=event_list.index(str(event_name))
+    cols=event_depth
+    lines=get_search_list(pipe_list,event_name)
+    for i in range(rows):
+      for j in range(cols):
+        chunks[index][i][j]=lines[(i*cols)+j]
```

Comment:
You can write this nested for loops in one line:
chunks[index] = [[lines[i * cols + j] for j in range(cols)] for i in range(rows)]

Reply:
fixed